lib/src/events.dart

```diff
@@ -207,62 +205,56 @@ class PresenceStatsResult {
 }
 
 class JoinEvent {
-  JoinEvent(this.user, this.client);
+  JoinEvent(this.info);
 
-  final String user;
-  final String client;
+  final proto.ClientInfo info;
```

Comment:
Personally, I think it is really important to maintain compatibility, but the interface unity of the external output SDK is also very critical. In this SDK, client and user are different from other languages. It feels like it is more superfluous, so after referring to other After installing the sdk, I decided to delete it directly instead of simply adding an info

Reply:
I mean events should be like this:

```
class JoinEvent {
  ...

  final String client;
  final String user;
  final List<int>? connInfo;
  final List<int>? chanInfo;
  
  ...
}  
```

This will allow us releasing new version of centrifuge-dart and not break other users' code upon upgrade. We can break compatibility if there is enough reasoning for it - but in this case it seems we can simply extend current events with new optional fields from `ClientInfo`. Or having `final ClientInfo info;` simplifies sth for your use case to justify breaking change at this point?

BTW: you are exposing `proto.ClientInfo` now, but we have another `ClientInfo` in `events.dart` to not export Protobuf structures to library users directly. See [ClientInfo](https://github.com/centrifugal/centrifuge-dart/blob/e03111c41d3f4c5badee4c1aa5f60e303ce7406a/lib/src/events.dart#L123)

Reply:
OK, cool, I'll repush later...